test/functional/ui/cursor_spec.lua

```diff
@@ -18,138 +18,155 @@ describe('ui/cursor', function()
   end)
 
   it("'guicursor' is published as a UI event", function()
-    local expected_cursor_style = {
-      cmdline_hover = {
-        mouse_shape = 0,
-        short_name = 'e' },
-      cmdline_insert = {
+    local expected_mode_info = {
+      [1] = {
         blinkoff = 250,
         blinkon = 400,
         blinkwait = 700,
-        cell_percentage = 25,
-        cursor_shape = 'vertical',
+        cell_percentage = 0,
+        cursor_shape = 'block',
+        full_name = 'normal',
         hl_id = 46,
         id_lm = 47,
         mouse_shape = 0,
-        short_name = 'ci' },
-      cmdline_normal = {
+        short_name = 'n' },
+      [2] = {
         blinkoff = 250,
         blinkon = 400,
         blinkwait = 700,
         cell_percentage = 0,
         cursor_shape = 'block',
+        full_name = 'visual',
         hl_id = 46,
         id_lm = 47,
         mouse_shape = 0,
-        short_name = 'c' },
-      cmdline_replace = {
+        short_name = 'v' },
+      [3] = {
         blinkoff = 250,
         blinkon = 400,
         blinkwait = 700,
-        cell_percentage = 20,
-        cursor_shape = 'horizontal',
+        cell_percentage = 25,
+        cursor_shape = 'vertical',
+        full_name = 'insert',
         hl_id = 46,
         id_lm = 47,
         mouse_shape = 0,
-        short_name = 'cr' },
-      insert = {
+        short_name = 'i' },
+      [4] = {
         blinkoff = 250,
         blinkon = 400,
         blinkwait = 700,
-        cell_percentage = 25,
-        cursor_shape = 'vertical',
+        cell_percentage = 20,
+        cursor_shape = 'horizontal',
+        full_name = 'replace',
         hl_id = 46,
         id_lm = 47,
         mouse_shape = 0,
-        short_name = 'i' },
-      more = {
-        mouse_shape = 0,
-        short_name = 'm' },
-      more_lastline = {
-        mouse_shape = 0,
-        short_name = 'ml' },
-      normal = {
+        short_name = 'r' },
+      [5] = {
         blinkoff = 250,
         blinkon = 400,
         blinkwait = 700,
         cell_percentage = 0,
         cursor_shape = 'block',
+        full_name = 'cmdline_normal',
```

Comment:
The decision was made in https://github.com/neovim/neovim/pull/6044 (you can search the page for justin's answer:

> Yes. It's more a matter of not wanting to expose the internal order to the API, and then having to support those numbers forever more. It also is hopefully a better experience for API users.